core/gui/src/app/common/util/json.ts

```diff
@@ -20,26 +20,32 @@ export function isBinary(str: string): boolean {
   return binaryRegex.test(str);
 }
 
-export function trimDisplayJsonData(rowData: IndexableObject, maxLen: number): Record<string, unknown> {
-  return deepMap<Record<string, unknown>>(rowData, value => {
-    if (typeof value === "string") {
-      if (isBase64(value) || isBinary(value)) {
-        const length = value.length;
-        // If length is less than 13, show the entire string.
-        if (length < 13) {
-          return `bytes'${value}' (length: ${length})`;
-        }
-        // Otherwise, show the leading and trailing bytes with ellipsis in between.
-        const leadingBytes = value.slice(0, 10);
-        // If the length of the value is less than 10, leadingBytes will take the entire string.
-        const trailingBytes = value.slice(-3);
-        // If the length of the value is less than 3, trailingBytes will take the entire string.
-        return `bytes'${leadingBytes}...${trailingBytes}' (length: ${length})`;
-      }
-      if (value.length > maxLen) {
-        return value.substring(0, maxLen) + "...";
-      }
+export function formatBinaryData(value: string): string {
+  const length = value.length;
+  // If length is less than 13, show the entire string.
+  if (length < 13) {
+    return `bytes'${value}' (length: ${length})`;
+  }
+  // Otherwise, show the leading and trailing bytes with ellipsis in between.
+  const leadingBytes = value.slice(0, 10);
+  // If the length of the value is less than 10, leadingBytes will take the entire string.
+  const trailingBytes = value.slice(-3);
+  // If the length of the value is less than 3, trailingBytes will take the entire string.
+  return `bytes'${leadingBytes}...${trailingBytes}' (length: ${length})`;
+}
+
+export function trimAndFormatData(value: any, maxLen: number): string {
+  if (typeof value === "string") {
+    if (isBase64(value) || isBinary(value)) {
+      return formatBinaryData(value);
+    }
+    if (value.length > maxLen) {
+      return value.substring(0, maxLen) + "...";
     }
-    return value;
-  });
+  }
+  return value?.toString() ?? "";
```

Comment:
this would convert all values to string right? I think we need to handle null differently.

Reply:
I'll display it as `NULL` for the null value.